src/completion.rs

```diff
@@ -367,10 +367,14 @@ trait Completable {
 //
 // It is assumed that the haystack is the name of an identifier and the needle is a partial
 // identifier.
+#[allow(clippy::unwrap_used)]
 fn fuzzy_match(haystack: &str, needle: &str) -> bool {
-    return haystack
-        .to_lowercase()
-        .contains(needle.to_lowercase().as_str());
+    haystack.starts_with(needle)
+        || haystack.contains({
+            let mut chars: Vec<char> = needle.chars().collect();
+            chars[0] = chars[0].to_uppercase().next().unwrap();
+            chars.into_iter().collect::<String>().as_str()
```

Comment:
Nit: This does it with one allocation (per match)
```rust
needle.chars().take(1).map(char::to_uppercase).chain(needle.chars().skip(1)).collect::<String>().as_str()
```
or maybe we want to pull in a fuzzy matching library to make this less ad-hoc? Doesn't seem to be any tried and true projects for that though. https://crates.io/crates/fuzzy-matcher exists 🤷

Reply:
This new matching ends up case sensitive for any character except the first which is more restrictive than before.

For example `chandemom` won't match `chandeMomentumOscillator` anymore.

Reply:
> I think it is fair to expect that you can type your "needle" without taking case into account.

Yeah, I think that's absolutely a good point. I wonder if that needle needs be of sufficient length before we start trying to match on the _inside_ of the string. Lemme give it some thought.